streams/resources/rs-utils.js

```diff
@@ -177,9 +177,21 @@
     return stream;
   }
 
+  function transferArrayBufferView(view) {
+    const noopByteStream = new ReadableStream({
+      type: 'bytes',
+      pull(c) {
+        c.byobRequest.respond(c.byobRequest.view.byteLength);
+        c.close();
+      }
+    });
+    const reader = noopByteStream.getReader({ mode: 'byob' });
+    return reader.read(view).then((result) => result.value);
+  }
```

Comment:
This was the "easiest" solution I could come up with to transfer an `ArrayBufferView`.

We *could* also use `postMessage()` like we do with [createTransferredReadableStream](https://github.com/web-platform-tests/wpt/blob/f2eb8b9fa0d0d11429ce4952dbb9dd796c18e0da/streams/transferable/resources/helpers.js#L85-L102), but then we also need to change the test setup for the reference implementation... and I can't be bothered. 😛

Reply:
enqueue-with-detached-buffer.window.js also uses postMessage, hmm. It might be worth changing https://github.com/whatwg/streams/blob/0ff6d4508b58acce28386696b2854b2a1570d205/reference-implementation/run-web-platform-tests.js#L43-L56 to just stick the "transferred" flag onto the transferred argument.

Reply:
You mean replacing `window.postMessage` to add our special "transferred" flag to every `ArrayBuffer` in `transferList`? Not sure how feasible that would be...
* We would need to export the [TransferArrayBuffer](https://github.com/whatwg/streams/blob/0ff6d4508b58acce28386696b2854b2a1570d205/reference-implementation/lib/abstract-ops/ecmascript.js#L16-L31) helper in [index.js](https://github.com/whatwg/streams/blob/0ff6d4508b58acce28386696b2854b2a1570d205/reference-implementation/lib/index.js), so that we can access it [after loading the bundle](https://github.com/whatwg/streams/blob/0ff6d4508b58acce28386696b2854b2a1570d205/reference-implementation/run-web-platform-tests.js#L56). We *have* to use the same helper, so that it'll use the exact same [isFakeDetached](https://github.com/whatwg/streams/blob/0ff6d4508b58acce28386696b2854b2a1570d205/reference-implementation/lib/abstract-ops/ecmascript.js#L4) symbol.
* When we override `window.postMessage(message, targetOrigin, transfer)`, we have to transfer any `ArrayBuffer`s inside `transfer` *and* replace all references to those buffers inside `message`. Sure, we can *assume* that for our tests, the message will always be an `ArrayBufferView` (and not some deeply nested JavaScript object).

Sure, I guess I can try it. Not sure if it'll be pretty though. 😛 (It's times like these where I wish [tc39/proposal-resizablearraybuffer](https://github.com/tc39/proposal-resizablearraybuffer) were already a reality... 😞)

Reply:
To be clear, I'm not sure it's worth it; just an idea to throw out there!

Reply:
All right, so this would work:
```javascript
const windowPostMessage = window.postMessage.bind(window);
window.postMessage = function (message, targetOrigin, transfer) {
  if (targetOrigin === '*' && transfer.length === 1 && transfer[0] instanceof window.ArrayBuffer) {
    const originalBuffer = transfer[0];
    const transferredBuffer = window.whatwgStreamsInternals.TransferArrayBuffer(originalBuffer);
    if (window.ArrayBuffer.isView(message) && message.buffer === originalBuffer) {
      message = new message.constructor(transferredBuffer, message.byteOffset, message.byteLength);
    }
    transfer = [transferredBuffer];
  }
  windowPostMessage(message, targetOrigin, transfer);
};
```
`window.whatwgStreamsInternals` would be defined in `lib/index.js` as:
```javascript
// ...
const { TransferArrayBuffer } = require('./abstract-ops/ecmascript.js');
window.whatwgStreamsInternals = { TransferArrayBuffer };
```
Then we can change `transferArrayBufferView` inside the test utilities:
```javascript
function transferArrayBufferView(view) {
  return new Promise((resolve) => {
    self.addEventListener('message', (event) => resolve(event.data), { once: true });
    self.postMessage(view, '*', [view.buffer]);
  });
}
```
Do we like this better? I think it looks nicer in WPT, but the runner becomes a little bit more complicated. That said, many different browser developers need to read and understand the WPT tests, so I suppose we should favor simplicity of the tests over simplicity of our test runner.

Reply:
I prefer the version that doesn't involve us monkey patching postMessage in the reference implementation. The weirdness of the byte-stream-based `transferArrayBufferView()` implementation is nicely encapsulated, so it's not hurting anyone.